types/usb/index.d.ts

```diff
@@ -404,7 +424,7 @@ export class InEndpoint extends EventEmitter implements Endpoint {
    * @param length
    * @param callback
    */
-  transfer(length: number, callback: (error: LibUSBException, data: Buffer) => void): InEndpoint;
+  transfer(length: number, callback: (error?: LibUSBException, data?: Buffer) => void): InEndpoint;
```

Comment:
This should always return an error object?

Reply:
thanks, fixing it

Reply:
wait no, I don't see the issue here :thinking: I'm changing the callback to make the parameters optional, is that incorrect?

Reply:
I believe functions follow the [error first callback pattern](https://nodejs.org/api/errors.html#errors_error_first_callbacks), so the error param cannot be optional as it's used to pass null or undefined in the case it succeeds.

Reply:
e.g. [in this case](https://github.com/thegecko/webusb/blob/0a3f33156129f9e042ede361aab69164b0a00588/src/adapter.ts#L752) the error param could be the data by your definition, but in reality error is always passed.

Reply:
oh, I see... would `(error: LibUSBException | undefined, data?: Buffer)` be better?

Reply:
absolutely!

Reply:
fixed!